### ISSUE TITLE
Open Source Support

### ISSUE THREAD

**Issue by alexdryden** [pressbooks/pressbooks#3759]
I wanted to check in to see what your plan are with respect to open source support. I have noticed over the past few years that some components of the Pressbooks ecosystem have become closed source and other elements have become obsoleted and replaced with closed source alternatives. Obviously, given your support model, you need to consider your enterprise business as your top priority. My concern is that as more components become part of the enterprise-only system, it may become increasingly difficult to maintain an open-source version of the software in parallel. We were a little blindsided in the past when some themes were discontinued, for example. 

We are coming up on an evaluation period that includes reviewing some of our technical infrastructure. There are three questions I'd like to be able to answer during our review, that maybe you can help with:
- What kind of bandwidth do you anticipate having available for community driven features or pull request?
- Do your CI/CD pipelines test an OSS only version? Basically, could we pull down an update that hasn't been tested against an open source implementation?
- Will public facing theming guides be maintained? If we use a custom theme, will we be able to anticipate breaking changes?

Any insights that you can give that might help me quantify the risk we face from continued investment in Pressbooks as an open source tool would be really appreciated. 

(Tagging @dgtracy who will be evaluating our tech stack.)

**Comment by cmurtagh** [comment]
Greetings @alexdryden. Thank you for these great questions. We do strive to maintain a well-working product for our open source users, and as part of that effort we're working on a couple of new things (some of these are in progress and/or in our backlog):

1) We're updating how we support devs with new documentation on how to setup local development using more modern tools (a.k.a. containers and not VirtualBox). Our devs are currently using this and we will be updating our public facing docs soon.

2) We're setting up an all OSS stack within our AWS infrastructure that we will use for manual and automated testing. Our current CI/CD does run tests against our repos/plugins, including the OSS content and we do try to minimize the non-open parts. These tests should cover OSS only users, but the new OSS only infra should help give us more. We're happy to chat with OSS users on how we can improve that and collaborate on making this better.

3) We're creating a better way for open source deploys using composer and Bedrock (from Roots.io) that is far more stable and easier to deploy. This will eventually deprecate any other install methods. 

We also welcome PRs and community driven feature requests. We recognize that we have been slow to adopt some of these in the past, but we are committed to improve that (we're a small team in an EDU market, which does put greater resource constraints than we would want sometimes).  If you have any suggestions or requests on what we might want to prioritize, please let us know with a comment in this issue.

**Comment by alexdryden** [comment]
Thanks @cmurtagh, it is really helpful to know that you are setting up an all OSS stack that will get eyes and automated testing. And we are looking forward to the updated docs on your containerization and build solutions!

In terms of features that we would like to advocate for, our main unresolved need is a way to display other authorship models (e.g., edited collections, translations of historic texts, etc.) Much of that discussion has been captured here https://github.com/pressbooks/ideas/issues/134#issuecomment-944830722 and in these pull requests https://github.com/pressbooks/pressbooks-book/pull/981 https://github.com/pressbooks/pressbooks/pull/2657. 

I am definitely willing to contribute and to work to adapt a pr to help satisfy an existing need, especially if that can help you justify the time commitments required to oversee a community pr. And I also understand that bandwidth issues mean that some priorities have to come before others. But because I am a solo developer supporting 5 open source platforms for our press, I need to prioritize my effort where it will have the greatest impact. If this feature is something you think could still have legs, I'd be happy to reengage.

**Comment by douglsmith** [comment]
As an open source user, I'll add that we made that choice to have full control over our content and tools. We'd love a way to pay for some of the closed-source add-ons to have those tools and to support the project financially. Also, if there are improvements in working with the community on PRs we would also likely participate there as well.

**Comment by cmurtagh** [comment]
@douglsmith Thank you for your feedback. We're definitely looking for ways to be more engaging with our open source users. It's a slow process (we're a small team in a market that is rapidly changing, especially in the current political climate). I'm not sure how we would be able to sell our SaaS client-only features, but we do want our open source users to have the best experience possible as well. Please feel free to reach out to us directly when you can if you have ideas and suggestions on how we can improve that.

**Comment by douglsmith** [comment]
Obviously, there are some SaaS features that require your infrastructure to function. But how about offering something like the additional book themes that are not in the open-source version? I'd be happy to pay for that.

**Comment by SteelWagstaff** [comment]
@douglsmith @alexdryden We've made a few updates that should help open source users. These are still works in progress, but hopefully movement in the right direction:
1. We've created a repository that uses Lando/Docker to provision a local instance of Pressbooks for testing/development by OSS contributors: https://github.com/pressbooks/local-dev-environment. The setup steps should be easy to follow and 'just work'
2. We've just updated our docs for folks who want to self-host Pressbooks (https://pressbooks.org/user-guides/installation/) along with a sample Bedrock: https://github.com/pressbooks/pressbooksoss-bedrock?tab=readme-ov-file#installing-pressbooks-on-a-production-server. More planned improvement described here: https://github.com/pressbooks/pressbooks.org/issues/142#issuecomment-3363701443
3. We've also restructured our community forum -- there are now dedicated sections for ['Contributing to Pressbooks'] (https://pressbooks.community/c/contributing/12) and ['Installing/Hosting Pressbooks'](https://pressbooks.community/c/hosting/15)
4. We've set up an option to allow open source folks to 'Sponsor' Pressbooks on GitHub: https://github.com/sponsors/pressbooks

We're open to other suggestions and improvements that you and others would like to see and want to be better at encouraging contributions of all kinds (PRs, translations, sponsorship, etc.)

**Comment by douglsmith** [comment]
Those are some great updates @SteelWagstaff. Thanks!
